api/src/OAuth/JWTStateOAuth2Client.php

```diff
@@ -69,14 +72,13 @@ public function redirect(array $scopes = [], array $options = []): RedirectRespo
 
         try {
             $response->headers->setCookie(
-                Cookie::create($this->getCookieName($this->apiDomain, false))
-                    ->withValue($this->encodeStateJWT(array_merge($options['additionalData'], [
+                Cookie::create($this->getCookieName($this->cookiePrefix))
+                    ->withValue($this->encodeStateJWT(array_merge($options['additionalData'] ?? [], [
                         'state' => $state,
                         'iat' => $now,
                         'exp' => $expires,
                     ])))
                     ->withHttpOnly()
-                    ->withDomain($this->apiDomain)
```

Comment:
If we remove the domain from the cookie, what is the fallback? It just uses the current request domain as the domain limiter? Why do we want to remove the domain?

Reply:
Exactly, it's the domain of the current request. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

Actually, when setting the domain option manually, subdomains of that domain are also automatically included. So omitting this option is actually more restrictive. See also the info on __Host- cookies on MDN